src/main/scala/io/iohk/ethereum/blockchain/data/GenesisDataLoader.scala

```diff
@@ -90,20 +94,22 @@ class GenesisDataLoader(
     import MerklePatriciaTrie.defaultByteArraySerializable
 
     val ephemDataSource = EphemDataSource()
-    val ephemNodeStorage = new NodeStorage(ephemDataSource)
+    val nodeStorage = new NodeStorage(ephemDataSource)
 
-    val initialStateMpt =
-      MerklePatriciaTrie[Array[Byte], Account](ephemNodeStorage, (input: Array[Byte]) => crypto.kec256(input))
-    val stateMpt = genesisData.alloc.foldLeft(initialStateMpt) { case (mpt, (address, AllocAccount(balance))) =>
+    val initalRootHash = MerklePatriciaTrie.calculateEmptyRootHash((input: Array[Byte]) => crypto.kec256(input))
+
+    val stateMpt = genesisData.alloc.zipWithIndex.foldLeft(initalRootHash) { case (rootHash, (((address, AllocAccount(balance)), idx))) =>
+      val ephemNodeStorage =  PruningMode.nodesKeyValueStorage(pruningMode, nodeStorage)(Some(idx - genesisData.alloc.size))
+      val mpt = MerklePatriciaTrie[Array[Byte], Account](rootHash, ephemNodeStorage, (input: Array[Byte]) => crypto.kec256(input))
       val paddedAddress = address.reverse.padTo(addressLength, "0").reverse.mkString
-      mpt.put(crypto.kec256(Hex.decode(paddedAddress)), Account(blockchainConfig.accountStartNonce, UInt256(BigInt(balance)), emptyTrieRootHash, emptyEvmHash))
+      mpt.put(crypto.kec256(Hex.decode(paddedAddress)), Account(0, UInt256(BigInt(balance)), emptyTrieRootHash, emptyEvmHash)).getRootHash
```

Comment:
Why does the nonce of the initial accounts be always 0? Maybe I'm missing something but it seems from [parity morden configuration](https://github.com/paritytech/parity/blob/master/ethcore/res/ethereum/morden.json#L62) that the starting nonce also applies to the initial accounts.